src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePickerSelector.iOS.cs

```diff
@@ -4,84 +4,74 @@
 using Foundation;
 using UIKit;
 using Uno.UI.Extensions;
+using Uno.Extensions;
+using Uno.Logging;
 
 namespace Windows.UI.Xaml.Controls
 {
 	public partial class DatePickerSelector
 	{
 		private UIDatePicker _picker;
-		private readonly SerialDisposable _dateChangedSubscription = new SerialDisposable();
+		private NSDate _initialValue;
+		private NSDate _newValue;
 
 		protected override void OnLoaded()
 		{
 			base.OnLoaded();
 
-			_picker = this.FindSubviewsOfType<UIDatePicker>()
-				.FirstOrDefault();
-
-			var parent = _picker.FindFirstParent<FrameworkElement>();
-
-			if (_picker != null)
+			_picker = this.FindSubviewsOfType<UIDatePicker>().FirstOrDefault();
+			if (_picker == null)
 			{
-				_picker.Mode = UIDatePickerMode.Date;
-				_picker.TimeZone = NSTimeZone.LocalTimeZone;
-				_picker.Calendar = new NSCalendar(NSCalendarType.Gregorian);
-				_picker.SetDate(Date.Date.ToNSDate(), animated: false);
-
-				//Removing the date picker and adding it is what enables the lines to appear. Seems to be a side effect of adding it as a view. 
-				if (parent != null)
-				{
-					parent.RemoveChild(_picker);
-					parent.AddSubview(_picker);
-				}
-
-				RegisterValueChanged();
+				this.Log().DebugIfEnabled(() => $"No {nameof(UIDatePicker)} was found in the visual hierarchy.");
```

Comment:
This is an error, and in general you must not use DebugIfEnabled (it creates a delegate for no good reason).

Reply:
So, this pattern is preferred instead?
```
if (this.Log().IsEnabled(LogLevel.Error))
{
	this.Log().Error("Exception", e);
```

Reply:
They are from the existing code, I just moved them around. I will upgrade them to errors.